network-observability.md

```diff
@@ -0,0 +1,84 @@
+
+There is no perfect match between OTEL semantics and NetObserv format, however several conventions can be extrapolated to create a new model in OTEL style:
+
+- Traffic by source and destination: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/attributes.md#source-and-destination-attributes
+- Kubernetes resources: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/k8s.md
+
+These two specifications are combined by prefixing with "source." or "destination." any resource such as a Kubernetes Pod.
+
+A tradeoff is also done to avoid too much redundancy and its associated cost (increased message size, increased storage): the model being entirely related to networking, the `network.` prefix is omitted and considered as implicit. As a result, a field such as the network protocol is referred to as `protocol` instead of `network.protocol`.
+
+## Format proposal
+
+This new format is currently a proposal and is subject to change before being implemented in NetObserv.
+
+The conversions are as follow:
+
+```
+SrcAddr                -> source.address
+SrcMac                 -> source.mac
+SrcHostIP              -> source.host.address
+SrcK8S_HostName        -> source.k8s.node.name
+SrcPort                -> source.port
+SrcK8S_Name            -> source.k8s.name
+SrcK8S_Type            -> source.k8s.kind
+SrcK8S_OwnerName       -> source.k8s.owner.name
+SrcK8S_OwnerType       -> source.k8s.owner.kind
+SrcK8S_Namespace       -> source.k8s.namespace.name
+SrcK8S_HostIP          -> source.k8s.host.address
+SrcK8S_HostName        -> source.k8s.host.name
+SrcK8S_Zone            -> source.zone
+DstAddr                -> destination.address
+DstMac                 -> destination.mac
+DstHostIP              -> destination.host.address
+DstK8S_HostName        -> destination.k8s.node.name
+DstPort                -> destination.port
+DstK8S_Name            -> destination.k8s.name
+DstK8S_Type            -> destination.k8s.kind
+DstK8S_OwnerName       -> destination.k8s.owner.name
+DstK8S_OwnerType       -> destination.k8s.owner.kind
+DstK8S_HostIP          -> destination.k8s.host.address
+DstK8S_HostName        -> destination.k8s.host.name
+DstK8S_Namespace       -> destination.k8s.namespace.name
+DstK8S_Zone            -> destination.zone
+Bytes                  -> bytes
+Packets                -> packets
+Proto                  -> protocol
+Flags                  -> tcp.flags
+TimeFlowRttNs          -> tcp.rtt
+Interface              -> interface.name
+IfDirection            -> interface.direction
+FlowDirection          -> host.direction
+DnsErrno               -> dns.errno
+DnsFlags               -> dns.flags
+DnsFlagsResponseCode   -> dns.responsecode
+DnsId                  -> dns.id
+DnsLatencyMs           -> dns.latency
+Dscp                   -> dscp
+IcmpCode               -> icmp.code
+IcmpType               -> icmp.type
+K8S_ClusterName        -> k8s.cluster.name
```

Comment:
Since this field is not per `source` / `destination`, I feel it is more related to the `collector` than `k8s` enrichment.

We should either put it under `source` and `destination` with the ability to set different values between the two or add a new `collector` section that could also include AgentIP, TimeReceived etc

Reply:
Yes I see what you say, this field (as it is today implemented in netobserv) tells about the cluster that collects the flow, not the cluster from which flow starts or goes. Since this spec is (currently) designed to map what exists today in netobserv I'd rather go with the "collector" option ... which isn't mutually exclusive with adding later a src/dst cluster